File: ProblemE.py
''' 
 @ author :  AmirZoyber 
 language : Python3.* 
 ''' 
  
 def sort(endTime,startTime): 
     for i in range(len(endTime)): 
         for j in range(len(endTime)-i-1): 
             if (endTime[j]>endTime[j+1]): 
                 endTime[j],endTime[j+1] = endTime[j+1],endTime[j] 
                 startTime[j],startTime[j+1] = startTime[j+1],startTime[j] 
     return endTime,startTime 
  
 n = int(input()) 
 startTime = list(map(int,input().split())) 
 endTime= list(map(int,input().split())) 
  
 endTime,startTime=sort(endTime,startTime) 
  
 sum=1;count=0 
  
 for i in range(1,n): 
     if(endTime[count] <= startTime[i]): 
         sum+=1 
         count=i 
 print(sum)

File: ProblemE.cpp
/**
 * @file ProblemFifth.cpp
 * @author @mahdimma
 * @brief
 * @version 0.1.0
 * @date 2022-08-28
 *
 * @copyright Copyright (c) 2022
 *
 */

#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

vector<pair<int, int>> vect;
int startTime[1000];

int main()
{
  int N;
  cin >> N;
  for (int i = 0; i < N; i++)
  {
    cin >> startTime[i];
  }
  int endTime = 0;
  for (int i = 0; i < N; i++)
  {
    cin >> endTime;
    vect.push_back(make_pair(endTime, startTime[i]));
  }
  sort(vect.begin(), vect.end());
  int sum = 1;
  int endT = vect[0].first;
  for (int i = 1; i < N; i++)
  {
    if (endT <= vect[i].second)
    {
      endT = vect[i].first;
      sum++;
    }
  }
  cout << sum;
}